### PR TITLE
Add delete modal to template info page

### PR DESCRIPTION
**Pull request by gracez72**
[TASK-34081](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/Adaptive%20Cards/Intern%20GitHub/Y20-VW/09?workitem=34081), [TASK-33734](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/Adaptive%20Cards/Intern%20GitHub/Y20-VW/09?workitem=33734)

This PR allows deleting live versions of templates and adds the delete modal to the template info page. 

This PR also fixes the delete bug in the client SDK and recompiled the client SDK. The files under `adaptivecards-templating-client/typescript` were autogenerated by the changes in the `swagger.yaml` file. 

**Screenshot**: 
delete

**Mocks**: 
No explicit mocks, followed design of publish modal.

### CODE REVIEW

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  render() {
+    const { template } = this.props;
+
+    return (
+      <BackDrop>
+      <Modal>
+        <Header>Delete Template</Header>
+        <Description style={{ marginBottom: 0 }}>{STRINGS.DELETE_CONFIRMATION}<DescriptionAccent>{template.name} - {this.props.templateVersion}</DescriptionAccent>?</Description>
```
Comment (daniellemott) [inline private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: Can styling be added to styled.tsx folder?

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        <CenterPanelWrapper>
+          <Container>
+            <ACWrapper>
+              <AdaptiveCard cardtemplate={template} templateVersion={this.props.templateVersion} />
+            </ACWrapper>
+          </Container>
+        </CenterPanelWrapper>
+        <BottomRow>
+          <ButtonGroup>
+            <CancelButton text="Cancel" onClick={this.props.closeModal} />
```
Comment (daniellemott) [inline private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: Add string text to string assets file

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 3]
```diff
@@ -14,1487 +14,1343 @@ import localVarRequest = require('request');
...
-        instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
-      }
-      return instance;
+
+    public static serialize(data: any, type: string) {
+        if (data == undefined) {
+            return data;
+        } else if (primitives.indexOf(type.toLowerCase()) !== -1) {
+            return data;
+        } else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
```
Comment (turalf) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: are those just a formatting chaneges?

File: private-templates-service/adaptivecards-templating-client/swagger.yaml [hunk 4]
```diff
@@ -356,7 +349,7 @@ definitions:
       _id: 
         type: string
       json: 
-        type: string
+        type: object
```
Comment (daniellemott) [inline private-templates-service/adaptivecards-templating-client/swagger.yaml]: Line 430: should type should be object as well?

File: private-templates-service/adaptivecards-templating-service/src/tests/router.test.ts [hunk 5]
```diff
@@ -9,32 +9,33 @@ import { InMemoryDBProvider } from "../storageproviders/InMemoryDBProvider";
...
 
 export default async function getToken(): Promise<string> {
-  const request = require("request-promise");
-  const endpoint = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/oauth2/token";
-  const requestParams = {
-    grant_type: "client_credentials",
-    client_id: "#{CLIENT_ID_TOKEN}#",
-    client_secret: "#{CLIENT_SECRET_TOKEN}#",
-    resource: "#{CLIENT_ID_TOKEN}#"
-  };
```
Comment (turalf) [inline private-templates-service/adaptivecards-templating-service/src/tests/router.test.ts]: is this for testing? 
I would not recommend checking in commented code

Comment (gracez72) [reply private-templates-service/adaptivecards-templating-service/src/tests/router.test.ts]: My mistake that was accidentally added, fixed!

File: private-templates-service/adaptivecards-templating-client/swagger.yaml (hunk 4 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/swagger.yaml]: For the authId?

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx (hunk 1 above)
Comment (gracez72) [reply private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: Fixed, thanks!

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx (hunk 2 above)
Comment (gracez72) [reply private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: Fixed!

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts (hunk 3 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: Yep, the files under `adaptivecards-templating-client/typescript` were autogenerated from the `swagger.yaml` file, I've updated the description to include this as well!

File: private-templates-service/adaptivecards-templating-client/swagger.yaml (hunk 4 above)
Comment (daniellemott) [reply private-templates-service/adaptivecards-templating-client/swagger.yaml]: My bad, thought those were templates
image

Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/swagger.yaml]: Oh okay I see for the recently edited/viewed templates? I'll put up a separate PR to fix this, might need to also edit the recent calls

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 6]
```diff
@@ -14,1487 +14,1343 @@ import localVarRequest = require('request');
 import http = require('http');
 import Promise = require('bluebird');
 
-let defaultBasePath = 'http://localhost:5000';
+let defaultBasePath = 'https://adaptivecms.azurewebsites.net';
```
Comment (daniellemott) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: Was this on purpose?

Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/typescript/api.ts]: Nope, fixed base path in swagger file, thanks for the catch!

File: private-templates-service/adaptivecards-templating-client/swagger.yaml [hunk 7]
```diff
@@ -6,7 +6,7 @@ info:
   license:
     name: "Apache 2.0"
     url: "http://www.apache.org/licenses/LICENSE-2.0.html"
-host: "adaptivecms.azurewebsites.net"
+host: "localhost:5000"
```
Comment (kodyang) [inline private-templates-service/adaptivecards-templating-client/swagger.yaml]: is this intentional

File: private-templates-service/adaptivecards-templating-client/typescript/api.ts [hunk 8]
```diff
@@ -22,1479 +22,1335 @@ let defaultBasePath = 'http://localhost:5000';
```
Comment (kodyang) [inline private-templates-service/adaptivecards-templating-client/typescript/api.ts]: from now on can you disable your linter

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  delete = () => {
+    this.props.deleteTemplate(this.props.templateVersion);
+    this.props.closeModal();
+  }
+
+  render() {
+    const { template } = this.props;
+
+    return (
+      <BackDrop>
```
Comment (kodyang) [inline private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: extra tab here

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+      <Modal>
+        <Header>Delete Template</Header>
```
Comment (kodyang) [inline private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: move this into strings file

File: private-templates-service/client/src/components/Common/DeleteModal/styled.tsx [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import styled from 'styled-components';
+
+export const ConfirmationDescription = styled.div`
+  margin-bottom: 0px;
```
Comment (kodyang) [inline private-templates-service/client/src/components/Common/DeleteModal/styled.tsx]: Not needed, `div` automatically has `margin-bottom: 0px` i believe

File: private-templates-service/adaptivecards-templating-client/swagger.yaml (hunk 7 above)
Comment (gracez72) [reply private-templates-service/adaptivecards-templating-client/swagger.yaml]: Yep, we replace it in the pipeline depending on which env

File: private-templates-service/client/src/components/Common/DeleteModal/styled.tsx (hunk 11 above)
Comment (gracez72) [reply private-templates-service/client/src/components/Common/DeleteModal/styled.tsx]: Fixed!

File: private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx (hunk 10 above)
Comment (gracez72) [reply private-templates-service/client/src/components/Common/DeleteModal/DeleteModal.tsx]: Fixed!

**Pull request merged**